assets/index.js

```diff
@@ -0,0 +1,138 @@
+
+	// Consts for the actual dropdown element
+	const searchDropdown = document.getElementById("searchDropdown");
+	const versionDropdown = document.getElementById("versionDropdown");
+
+	// Const for navbar checkbox
+	const navState = document.getElementById("nav-state")
+
+	let targetElement = e.target;
+
+	do {
+		if (targetElement == searchDropdownArea) {
+			// If clicked area is in search dropdown remove all other dropdowns
+			versionDropdown.classList.remove("show");
+			return;
+		} else if (targetElement == versionDropdownArea) {
+			// If clicked area is in version dropdown remove all other dropdowns
+			searchDropdown.classList.remove("show");
+			return;
+		} else if (targetElement == navArea) {
+			// Because this is ran after the other two check it is okay to close all dropdowns
+			// even through they are contained in nav.
+			versionDropdown.classList.remove("show");
+			searchDropdown.classList.remove("show");
+			return;
+		}
+		targetElement = targetElement.parentNode;
+	} while (targetElement);
+
+	// If clicked area is outside all dropdowns remove all other dropdowns
+	versionDropdown.classList.remove("show");
+	searchDropdown.classList.remove("show");
+
+	// Close dropdown on mobile
+	navState.checked = false;
+
+});
+
+// Search with specific search provider
+const searchDocsWithProvider = (searchProvider) => {
+	if (searchProvider === "google") {
+		location.href = 'https://www.google.com/search?ie=UTF-8&q=site%3Adocs.sourcegraph.com+ '+encodeURIComponent(document.getElementById('search').value);
+		return false;
+	} else if (searchProvider === "duckduckgo") {
+		location.href = 'https://www.duckduckgo.com/site%3Adocs.sourcegraph.com '+encodeURIComponent(document.getElementById('search').value);
+		return false;
+	} else if (searchProvider === "sourcegraph") {
+		location.href = 'https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph+file:doc/+file:%5C.md%24+'+encodeURIComponent(document.getElementById('search').value);
+		return false;
+
+	} else {
+		console.error("searchProvider not valid")
+	}
+
+}
+
+// Open and close nav section
+const toggleNavSection = (navSection) => {
+	const section = document.getElementById(navSection);
+	section.classList.toggle("expanded");
+}
+
+// Only open nav section
+const openNavSection = (navSection) => {
+	const section = document.getElementById(navSection);
+	if (section.classList.contains('expanded')) {
+		// Section is already expanded
+		return;
+	} else {
+		section.classList.add("expanded");
+		return;
+	}
+}
+
+// Open nav section based off current page
+const breadcrumbNavToggle = () => {
+	// Check if breadcrumb is defined
+	if (breadcrumbs.length === 0) {
+		// If breadcrumb isn't defined open the user nav section (on by default)
+		toggleNavSection('contentNavUser');
+		return;
+	}
+
+	var currentCatagory = breadcrumbs[1].Label;
+
+	// Open nav pannel based off current catagory
+	switch(currentCatagory) {
```

Comment:
I see some code here is not formatted - could you run prettier over everything so the next person to edit won't create unneeded diffs?

Reply:
Fixed in 
https://github.com/sourcegraph/sourcegraph/pull/2564/commits/bb4ce2e5486108dffa761285a4f9b53cae8f7853